### ISSUE TITLE
Ability to install extensions in a new profile from CLI

### ISSUE THREAD

**Issue by sandy081** [microsoft/vscode#172046]
Ability to install extensions in a new profile from CLI

At present this is not supported because CLI does not support creating profiles. But this technical limitation shall be fixed.

**Comment by sandy081** [comment]
@bpasero This is because the state service is not writeable in CLI. May I know if we can make this writeable in CLI?

**Comment by bpasero** [comment]
@sandy081 didn't we solve this for the shared process by providing a scoped storage service? Maybe we can do the same for CLI. The worry is the same: preventing multiple processes from accessing the same SQLite DB. Note that our SQLite DB is not configured to be written to from multiple processes.

**Comment by sandy081** [comment]
@bpasero To be clear its not storage, it is state service where we store the state in the json file 

https://github.com/microsoft/vscode/blob/e06874c041d367c00da10e6034fc2db4af135ec2/src/vs/platform/state/node/state.ts#L10

**Comment by bpasero** [comment]
@sandy081 ah, similar I guess. The `StateMainService` flushes its memory state to disk with delays and debouncing, so if one process changes the same file, its possible the changes get overwritten by the first process. In other words, the state service does not listen to file events or anything like that to reconcile.

**Comment by sandy081** [comment]
Right, is there a possibility for the CLI process to use State service from main if main is available otherwise use writeable state service in CLI?

**Comment by bpasero** [comment]
We have a way to talk to an existing instance if it runs (for example `ILaunchMainService` - this service is used when a second instance needs to talk to the first instance), so writing to state should be possible as well in that case.

**Comment by pablocabrera85** [comment]
Is there a way to help this get prioritised?

**Comment by AndrewDant** [comment]
I'm personally very disappointed to see this issue closed. I think this would be a very useful collaboration feature.
